lib/variable/reduction.cpp

```diff
@@ -228,8 +204,34 @@ void max_into(Variable &accum, const Variable &var) {
   accumulate_in_place(accum, var, core::element::max_equals, "max");
 }
 
+void nanmax_into(Variable &accum, const Variable &var) {
+  accumulate_in_place(accum, var, core::element::nanmax_equals, "max");
+}
+
 void min_into(Variable &accum, const Variable &var) {
   accumulate_in_place(accum, var, core::element::min_equals, "min");
 }
 
+void nanmin_into(Variable &accum, const Variable &var) {
+  accumulate_in_place(accum, var, core::element::nanmin_equals, "min");
+}
+
+/// Make a variable `accum` to pass to `sum_into(accum, data)`, etc.
+/// The variable is based on `data` and has the given dims.
+/// If `data` is binned, `accum` is dense with the elem dtype of `data`.
+Variable make_reduction_accumulant(const Variable &data,
+                                   const Dimensions &target_dims,
+                                   const FillValue &init) {
+  const auto type = variableFactory().elem_dtype(data);
+  const auto unit = variableFactory().elem_unit(data);
+  const auto has_variances = variableFactory().has_variances(data);
+  const auto scalar_prototype = empty(Dimensions{}, unit, type, has_variances);
+  return special_like(scalar_prototype.broadcast(target_dims), init);
+}
+
+void reduce_into(Variable &accum, const Variable &var,
+                 // cppcheck-suppress constParameter
```

Comment:
Why not passed as const instead?

Reply:
Because I don't know how. I couldn't find a place where the compiler would accept `const`.

Reply:
Switched to function *pointer*. Now I can add `const` and cppcheck is happy.